st2common/tests/unit/test_trigger_services.py

```diff
@@ -67,3 +67,63 @@ def test_create_trigger_db_from_rule_use_ref_for_trigger(self):
         trigger_db_ret_2 = trigger_service.create_trigger_db_from_rule(rule_api2)
         self.assertTrue(trigger_db_ret_2 is not None)
         self.assertTrue(trigger_db_ret_2.id == trigger_db_ret_1.id)
+
+    def test_create_or_update_trigger_db_simple_triggers(self):
+        test_fixtures = {
+            'triggertypes': ['triggertype1.json']
+        }
+        loader = FixturesLoader()
+        fixtures = loader.save_fixtures_to_db(fixtures_pack='generic', fixtures_dict=test_fixtures)
+        triggertypes = fixtures['triggertypes']
+        trigger_type_ref = ResourceReference.to_string_reference(
+            name=triggertypes['triggertype1.json']['name'],
+            pack=triggertypes['triggertype1.json']['name'])
+
+        trigger = {
+            'name': 'foo',
+            'pack': 'st2',
+            'type': trigger_type_ref
+        }
+        trigger_service.create_or_update_trigger_db(trigger)
```

Comment:
The name of the trigger should be the same as triggertype name if parameters are {}. I'll make that change. We just want to make sure that calling create_or_update with same trigger doesn't create new ones. The original code (before these changes) created duplicate.